### ISSUE TITLE
Some FEI emd EDS SI files cannot be opened

### ISSUE THREAD

**Issue by llajaunie** [hyperspy/hyperspy#1983]
Some EDS Si .emd FEI files aquired during the same TEM session can be opened in hyperspy whereas some others lead to the following error:

Example of file not working : https://files.fm/u/ms4bm9sb
```python
v2 = hs.load('SI EDS-HAADF-DF4-DF2-BF 1317.emd', select_type="spectrum_image")


---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-19-e074f81dc2d5> in <module>()
----> 1 v2 = hs.load('SI EDS-HAADF-DF4-DF2-BF 1317.emd', select_type="spectrum_image")

c:\users\usuario\hyperspy\hyperspy\io.py in load(filenames, signal_type, stack, stack_axis, new_axis_name, lazy, **kwds)
    264             objects = [load_single_file(filename, lazy=lazy,
    265                                         **kwds)
--> 266                        for filename in filenames]
    267 
    268         if len(objects) == 1:

c:\users\usuario\hyperspy\hyperspy\io.py in <listcomp>(.0)
    264             objects = [load_single_file(filename, lazy=lazy,
    265                                         **kwds)
--> 266                        for filename in filenames]
    267 
    268         if len(objects) == 1:

c:\users\usuario\hyperspy\hyperspy\io.py in load_single_file(filename, signal_type, **kwds)
    307                                 reader=reader,
    308                                 signal_type=signal_type,
--> 309                                 **kwds)
    310 
    311 

c:\users\usuario\hyperspy\hyperspy\io.py in load_with_reader(filename, reader, signal_type, **kwds)
    316     lazy = kwds.get('lazy', False)
    317     file_data_list = reader.file_reader(filename,
--> 318                                         **kwds)
    319     objects = []
    320 

c:\users\usuario\hyperspy\hyperspy\io_plugins\emd.py in file_reader(filename, log_info, lazy, **kwds)
   1149     if fei_check(filename) == True:
   1150         _logger.debug('EMD is FEI format')
-> 1151         emd = FeiEMDReader(filename, lazy=lazy, **kwds)
   1152         dictionaries = emd.dictionaries
   1153     else:

c:\users\usuario\hyperspy\hyperspy\io_plugins\emd.py in __init__(self, filename, select_type, first_frame, last_frame, sum_frames, sum_EDS_detectors, rebin_energy, SI_dtype, load_SI_image_stack, lazy)
    538             self._read_data(select_type)
    539         except Exception as e:
--> 540             raise e
    541         finally:
    542             if not self.lazy:

c:\users\usuario\hyperspy\hyperspy\io_plugins\emd.py in __init__(self, filename, select_type, first_frame, last_frame, sum_frames, sum_EDS_detectors, rebin_energy, SI_dtype, load_SI_image_stack, lazy)
    536                 self.p_grp = f.get('Presentation')
    537                 self._parse_image_display()
--> 538             self._read_data(select_type)
    539         except Exception as e:
    540             raise e

c:\users\usuario\hyperspy\hyperspy\io_plugins\emd.py in _read_data(self, select_type)
    567             _logger.info('Reading the spectrum image.')
    568             t0 = time.time()
--> 569             self._read_images()
    570             t1 = time.time()
    571             self._read_spectrum_stream()

c:\users\usuario\hyperspy\hyperspy\io_plugins\emd.py in _read_images(self)
    650         # each of them
    651         for image_sub_group_key in _get_keys_from_group(image_group):
--> 652             image = self._read_image(image_group, image_sub_group_key)
    653             if not self.load_images:
    654                 # If we don't want to load the images, we stop here

c:\users\usuario\hyperspy\hyperspy\io_plugins\emd.py in _read_image(self, image_group, image_sub_group_key)
    666             # if the `BinaryResult/Detector` is not available, there should be
    667             # only one detector in `Detectors`
--> 668             elif 'DetectorName' in original_metadata['Detectors']['Detector-01']:
    669                 self.detector_name = original_metadata['Detectors']['Detector-01']['DetectorName']
    670         except KeyError:

TypeError: string indices must be integers 
```

**Comment by thomasaarholt** [comment]
Which version of Velox is this from?

**Comment by llajaunie** [comment]
I think that's 2.3, this occurs for about 1/3 of my EDS SI.

**Comment by thomasaarholt** [comment]
I suggest downloading the new version 2.4, and either running the conversion tool or opening and saving them to get the emd files up to the newest version.

In velox, go Help - > About to find the download link for new versions.

Then give loading them into hyperspy another shot. May not work, but at least then we've eliminated that as a problem.

**Comment by francisco-dlp** [comment]
I am marking this as a bug because if this is a version issue we should:

* Document the minimum version of Velox required
* Raise a more useful message

**Comment by ericpre** [comment]
@llajaunie, can you confirm that it happen only with files containing EDS maps in addition to the spectrum image? 

The issue with the velox emd format is that it is not stable yet and we will have this kind of issue in the future. I think that we need to adjust the reader accordingly when we notice a change...

**Comment by ericpre** [comment]
The issue should be fixed in #1825.

**Comment by llajaunie** [comment]
@ericpre @thomasaarholt 
I am out of the lab without Velox, I will check these two points next week when I will be back.

**Comment by llajaunie** [comment]
@ericpre I can confirm that after "cleaning" the emd files (removing chemical maps, profile etc..), I am able to open the EDS SI in hyperspy.

**Comment by thomasaarholt** [comment]
@llajaunie if you want to avoid this in the future, you can either install the development version of hyperspy, or wait for the next (imminent) release.

**Comment by llajaunie** [comment]
@thomasaarholt 
I think I am working on the dev version, I have done the installation by using git and pip. Is there a way to check the version you are working on btw?

**Comment by thomasaarholt** [comment]
Absolutely:

```python
import hyperspy # not .api
hyperspy.__version__
```
Remember that in order to stay updated with the latest changes, you need to have done two things:
   1. Have installed hyperspy using `pip install -e .` from within your locally cloned version of hyperspy
   2. Recently have performed `git pull` from the hyperspy repository (not your own forked "origin").

In case you haven't done 1, do `pip uninstall hyperspy` and then do the `pip install -e .` command from within the local folder.

If you haven't done 2, run `git remote add upstream https://github.com/hyperspy/hyperspy.git`.
That creates the "upstream" remote, which allows you to then call `git pull upstream RELEASE_next_minor`, to get the most recent release.

**Comment by ericpre** [comment]
Fixed in #1825.
